### PR TITLE
fix: various issues preventing impersonated accounts and forks from working

### PR DESCRIPTION
**Pull request by antazoey**
### What I did

* fix: issue with static-fee txns where impersonated accounts failed without balance
* fix: issue where transactions state was lost for proxy-based accounts
* fix: forked config attr error
* fix: issue with 
* fix: issue where we would reset the fork to a problematic base fee

fixes: https://github.com/ApeWorX/ape/issues/1783

### How to verify it

1: set gas price in CLI to default to 0
2: delete hack in code that is no longer necessary (at least I think it is no longer necessary??)
 - So i am referring to the `set_code(0)` part for impersonated contracts we were doing... we had to do that before but it seems we no longer need to? also it seems to cause problems... (one of many, unfortunately)
3: set `self._fork_config.upstream_provider = upstream_name`
4: delete the part in reset fork where we do `self._make_request("anvil_setNextBlockBaseFeePerGas", [base_fee])` as that seems to cause problems and is no longer needed (again, is this actually no longer needed with this PR's changes?? because it just seems to cause problems otherwise that are fixed by deleting that  code).

### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

**Comment by antazoey** [comment]
**NOTE**: Tests are failing because they require this PR: https://github.com/ApeWorX/ape/pull/1814

File: ape_foundry/provider.py [hunk 1]
```diff
@@ -427,6 +431,8 @@ def build_command(self) -> List[str]:
             "--steps-tracing",
             "--block-base-fee-per-gas",
             f"{self.settings.base_fee}",
+            "--gas-price",
+            f"{self.settings.gas_price}",
```
Comment (fubuloubu) [inline ape_foundry/provider.py:435]: I believe you can't have both base fee and gas price at the same time

To get a more consistently nice testing environment, I think a gas price of zero works for local testing. For fork testing, I'm not sure if that'll work but it is better than using base fee because that always requires deducting gas

**Review by fubuloubu — APPROVED** [review]

File: ape_foundry/provider.py (hunk 1 above)
Comment (antazoey) [reply ape_foundry/provider.py:435]: i think it is just 1 settings applies to 1 type of tx and the other applies to another type of tx
also seems like brownie uses pre-1559 txns by default, which is how i got this bug... because i was trying to copy brownie in ape

Comment (fubuloubu) [reply ape_foundry/provider.py:435]: Yes, it went out of favor around the time EIP-1559 got popular

**Review by fubuloubu — APPROVED** [review]
